get-started-storage (#485)
opengisch/QField-docs: documentation/get-started/storage.en.md

tx_slug: documentation_get-started_storage_qfield
---

# QField Storage Management

On the QField homescreen, users are presented with two options to open a project.

>  **QFieldCloud projects**: The first option is to [access a project stored on QFieldCloud](../get-started/index.en.md)

Reviewer: ```suggestion
 **QFieldCloud projects**: The first option is to [access a project stored on QFieldCloud](../get-started/)
```

How to set up and work with QFieldCloud is explained in the next chapter.

* **Open local file**: The second option involves copying a working copy of the project file from a laptop or PC (the source device), on to the phone or tablet (the target device).
On the target device, users can access and edit this local copy using QField, and eventually copy the new version back to the desktop or to an alternative cloud storage service. How to prepare, store and access these local copies is the topic of this chapter.

There are various possibilities of exporting copies of project files and datasets from the source device runnign QGIS and importing them on to a target device for field-data collection. Below are OS-specific instructions on how to access projects and individual datasets through QField.

## Exporting QGIS project files for use in QField

> QFieldQField supports a [wide range of data formats](../reference/data-format).

Reviewer: ```suggestion
QFieldQField supports a [wide range of data formats](../reference/data-format.md).
```

> QFieldQField supports a [wide range of data formats](../reference/data-format/).

Reviewer: ```suggestion
QFieldQField supports a [wide range of data formats](../reference/data-format).
```

> QFieldQField supports a [wide range of data formats](../reference/data-format.en.md).

Reviewer: ```suggestion
QFieldQField supports a [wide range of data formats](../reference/data-format/).
```

There are three ways of preparing and copying a project in QGIS for use in QField.

* [Storing files in a designated folder](###storing-files-in-a-designated-folder)
* [Packaging the project as a GeoPackage](###packaging-the-project-as-a-geopackage)

> * [Packaging the project as a SpatiaLite database](###packaging-the-project-as-a-spatiaLite-database)

Reviewer 2: ```suggestion
```

tx_slug: documentation_get-started_storage_qfield
---

# QField Storage Management

> On the QField homescreen, users are presented with two options to open a project.
> 
> * **Open local file**: The first option involves copying a working copy of the project file from a laptop or PC (the source device), on to the phone or tablet (the target device).
> On the target device, users can accesses this local copy and manipulate it using QField, and eventually copy this new version back to the desktop or to an alternative cloud storage service. How to prepare, store and access these local copies is the topic of this chapter.
> 
> * **QFieldCloud projects**: The second option is to [access a project stored on QFieldCloud through QFieldSync.](#getting-started-guide)
> How to set up and work with QFieldCloud is explained in the next chapter.

Reviewer 2: ```suggestion
On the QField homescreen, users are presented with two options to open a project.

 **QFieldCloud projects**: The first option is to [access a project stored on QFieldCloud](../get-started/index.en.md)
How to set up and work with QFieldCloud is explained in the next chapter.

* **Open local file**: The second option involves copying a working copy of the project file from a laptop or PC (the source device), on to the phone or tablet (the target device).
On the target device, users can access and edit this local copy using QField, and eventually copy the new version back to the desktop or to an alternative cloud storage service. How to prepare, store and access these local copies is the topic of this chapter.

```

There are various possibilities of exporting copies of project files and datasets from the source device runnign QGIS and importing them on to a target device for field-data collection. Below are OS-specific instructions on how to access projects and individual datasets through QField.

!!! note
    Note to active QField users: Since November 2021, Google has, for security reasons, enforced a new set of storage access rules.
    These prohibit most applications from directly accessing files in many locations on a device‘s internal and external storages.
    Therefore, QField has to import projects and datasets into its Google-specified storage location within which QField has unrestricted read/write access.

>     Instructions below should help users' adjust to the new circumstances.

Reviewer 2: Can this not be deleted by now?

## Exporting QGIS project files for use in QField

> QFieldQField supports a [wide range of data formats](#supported-data-formats).
> There are three ways of preparing and copying a project in QGIS for use in QField.

Reviewer 2: ```suggestion
QFieldQField supports a [wide range of data formats](../reference/data-format.en.md).
There are three ways of preparing and copying a project in QGIS for use in QField.
```

* [Storing files in a designated folder](###storing-files-in-a-designated-folder)
* [Packaging the project as a GeoPackage](###packaging-the-project-as-a-geopackage)

> * [Packaging the project as a SpatiaLite database](###packaging-the-project-as-a-spatiaLite-database)

Reviewer 2: ```suggestion
```

### Storing files in a designated folder

One way of compiling all necessary parts of a QGIS project is to store them in a folder.
This folder can contain individual files such as a
* QGIS project file (.qgs or .qgz)
* Vector data (shapefiles, GeoJSON or Geopackage)
* Raster data (GeoTIFFs, JPEGS and others)
* Auxiliary files, which includes additional files used for styling (.qml or .sld) and any other files referenced by the project

### Packaging the project as a GeoPackage

> The most efficient way to package a QGIS project and its corresponding geospatial data into one file is to use [GeoPackages](https://www.geopackage.org/).
> We recommend using the GeoPackage file format for projects in QField, since packaging in QGIS is easy and straightforward. To package a project as a GeoPackage, three steps are necessary.

Reviewer 2: ```suggestion
The simplest and most efficient way to package a QGIS project and its corresponding geospatial data into one file is to use [GeoPackages](https://www.geopackage.org/).
We recommend using the GeoPackage file format for projects in QField, since packaging in QGIS is easy and straightforward. To package a project as a GeoPackage, three steps are necessary.
```

Reviewer 2: @SeqLaz  Is that correct ? I thought the .qgz file should be rather outside of the gpkg?

Reviewer 3: Yes, that is correct. It should be saved as a .qgs/.qgz file and not in a GeoPackage. The GeoPackage should be for the layers.

3. **Add project file (.qgs) to GeoPackage:**
Next go to Project -> Save to -> GeoPackage, and select the GeoPackage file containing all the vector layers. This saves the project file, with the .qgs extension into the GeoPackage as well.

<div style="padding:56.25% 0 0 0;position:relative;"><iframe src="https://player.vimeo.com/video/969846267?badge=0&amp;autopause=0&amp;player_id=0&amp;app_id=58479" frameborder="0" fullscreen; picture-in-picture; clipboard-write" style="position:absolute;top:0;left:0;width:100%;height:100%;" title="Package QGIS Project into GeoPackage including vector, raster and basemap layers (Copy)"></iframe></div><script src="https://player.vimeo.com/api/player.js"></script>
*Package QGIS Project into GeoPackage*

> For more information, consult the QGIS documentation for more information [packaging layers](https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/database.html#package-layers).

Reviewer 2: ```suggestion
For more information, consult the QGIS documentation  [packaging layers](https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/database.html#package-layers).
```

### Packaging the project as a SpatiaLite database

> Another possibility is to store the project and its layers in a [SpatiaLite](https://www.gaia-gis.it/fossil/libspatialite/index) database.

Reviewer 2: Is this recommended or should be avoided in which case I would probably mark that behind the sentence @SeqLaz

Reviewer 3: SpatiaLite is fully compatible. However, if possible, migrate the layers from SpatialLite to GeoPackage, as GeoPackage it is the recommended format for QField works!

Reviewer 4: I would not put it in the docs, this is leftover from the :sauropod: era.

After packaging the project accordingly, the folder or file containing the project has to be copied over to the target device running QField, specifically to the folder called `Imported Projects`.
This folder can be found on the following paths

* Android path: `Android/data/ch.opengis.qfield/files/Imported Projects`
* iOS path: `On My iPhone/QField/Imported Projects`

Copying the packaged project over to the corresponding folder of each device can be done in various ways.

### Android

#### Transfer via USB-cable

If you want to use a cable connection to import the project on to your android device, simply connect both devices using a USB cable and follow the instructions on [how to transfer files between your computer and the android device.](https://support.google.com/android/answer/9064445?hl=en#zippy=%2Cwindows-computer)
On most devices plugged into a computer via USB cable connection, the path will be `<drive>:/Android/data/ch.opengis.qfield/files/`.
Users will find both the Imported Datasets and Imported Projects folders within which the imported projects and datasets will be located.

> Changes done to project content and datasets are saved in the files found in this location.

Reviewer 2: ```suggestion
Changes done to the project content and datasets are saved in the files found in this location.
```

#### Google Drive (and other cloud storage services)

The advantage of using Google Drive is that both the source device and target device have shared access to a central working directory containing the current project files.
A possible workflow may look like this

* Prepare and package a QGIS project on the source device
* Upload the project to a cloud (For instance Google Drive)
* Download the project to the target devices and collect data
* Upload the changed project (or parts of the project) from the target devices back to the cloud and replace the old files with the new ones
* Download the project back on to the source device

!!! note
    When working with Google Drive, it may be helpful to create a dedicated folder in the cloud containing all the projects.
    A parallel of this folder can then also be created on the target device, to which the QGIS projects can be downloaded and save in.

### iOS

#### Transfer via USB-cable

> Transfering files from MacBooks or Macs to iPhone using a cable is not straightforward, since its not possible to access individual files in the QField directory.
> One workaround could be the following

Reviewer 2: ```suggestion
Transfering files from MacBooks or Macs to iPhone using a cable is not straightforward, since it is not possible to access individual files in the QField directory.
One workaround could be the following
```

After accessing and manipulating the project files in QField, use AirDrop on the target device to transfer the project files back to the source device.

## Importing projects and datasets

Apart from using QFieldCloud, QField can open projects and datasets in four ways:

* [importing a project folder](#importing-a-project-folder) <img src="../assets/images/android_robot.svg.png" alt="android" width="16" height="16"> <img src="../assets/images/apple_logo.svg.png" alt="apple" width="14" height="14">

* [importing a compressed project file](#importing-a-compressed-project) <img src="../assets/images/android_robot.svg.png" alt="android" width="16" height="16">

* [importing from a URL](#importing-from-a-url) <img src="../assets/images/android_robot.svg.png" alt="android" width="16" height="16"> <img src="../assets/images/apple_logo.svg.png" alt="apple" width="14" height="14">

* [importing individual datasets](#importing-individual-datasets) <img src="../assets/images/android_robot.svg.png" alt="android" width="16" height="16">

!![QField import actions](../assets/images/storage-import-actions.png)

> On Android all of these actions are available by click on the ‘import (+) button‘ located on the bottom-right corner of the project/dataset picker screen, which can be accessed by clicking on the ‘Open local files’ button located in QField‘s welcome screen.

Reviewer 2: ```suggestion
On Android all of these actions are available by clicking on the ‘import (+) button‘ located on the bottom-right corner of the project/dataset picker screen, which can be accessed by clicking on the ‘Open local files’ button located in QField‘s welcome screen.
```

tx_slug: documentation_get-started_storage_qfield
---

# QField Storage Management

On the QField homescreen, users are presented with two options to open a project.

 **QFieldCloud projects**: The first option is to [access a project stored on QFieldCloud](../get-started/index.en.md)
How to set up and work with QFieldCloud is explained in the next chapter.

* **Open local file**: The second option involves copying a working copy of the project file from a laptop or PC (the source device), on to the phone or tablet (the target device).
On the target device, users can access and edit this local copy using QField, and eventually copy the new version back to the desktop or to an alternative cloud storage service. How to prepare, store and access these local copies is the topic of this chapter.

There are various possibilities of exporting copies of project files and datasets from the source device runnign QGIS and importing them on to a target device for field-data collection. Below are OS-specific instructions on how to access projects and individual datasets through QField.

> !!! note

Reviewer 5: ```suggestion
```

>     Note to active QField users: Since November 2021, Google has, for security reasons, enforced a new set of storage access rules.
>     These prohibit most applications from directly accessing files in many locations on a device‘s internal and external storages.
>     Therefore, QField has to import projects and datasets into its Google-specified storage location within which QField has unrestricted read/write access.
>     Instructions below should help users' adjust to the new circumstances.

Reviewer 5: ```suggestion
```

2. **Add raster layers to GeoPackage**
If your project contains raster layers, then these can also be stored in the GeoPackage. Click on the raster layer you want to export, then “export” -> “save as”, and select GeoPackage as the format. Add a filename and select the three dots to browse for the GeoPackage. Select the GeoPackage and change the file format at the bottom of the dialogue window from “GeoTIFF” to “All files (*.*)”. Now both vector and raster layers are saved in the GeoPackage. To make sure, browse for the GeoPackage in the browser panel and expand it.

3. **Add project file (.qgs) to GeoPackage:**
Next go to Project -> Save to -> GeoPackage, and select the GeoPackage file containing all the vector layers. This saves the project file, with the .qgs extension into the GeoPackage as well.

![type:video](https://player.vimeo.com/video/969846267)<!-- markdown-link-check-disable-line -->
*Package QGIS Project into GeoPackage*

For more information, consult the QGIS documentation  [packaging layers](https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/database.html#package-layers).

> 
> ### Packaging the project as a SpatiaLite database
> 
> Another possibility is to store the project and its layers in a [SpatiaLite](https://www.gaia-gis.it/fossil/libspatialite/index) database.

Reviewer 5: ```suggestion
```
